src/sentry/profiles/task.py

```diff
@@ -523,19 +540,20 @@ def _insert_eventstream_profile(profile: Profile) -> None:
     """
 
     # just a guard as this should always be initialized already
-    if processed_profiles_publisher is None:
+    if _profiles_kafka_producer is None:
         return
 
-    processed_profiles_publisher.publish(
-        "processed-profiles",
-        json.dumps(profile),
+    f = _profiles_kafka_producer.produce(
+        Topic(name="processed-profiles"),
+        KafkaPayload(key=None, value=json.dumps(profile).encode("utf-8"), headers=[]),
     )
+    f.exception()
```

Comment:
Doesn't that mean this function is synchronous now? Before it was fire and forget and didn't need to wait for the future to be done. Now it blocks until the future is done in order to check whether there was an exception... Not sure if that was you intention?

Reply:
Yes, that's what it means and that was the intention. This is still to avoid losing profiles for no reason, if we push to Snuba and it doesn't publish for some reason, we lose the profile and we don't have an outcome for it.

Reply:
Ok, just keep in mind the throughput may be slower now. But you will have delivery guarantees.  In Arroyo we normally use threadpools so multiple messages can be produced at once.

Reply:
Right, this code is in the Celery task itself so it won't have an impact on the Arroyo consumer reading from Kafka. This consumer was kept single-threaded as it's already single-threaded.